api/api/http.js

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+'use strict';
+
+/** @namespace API-HTTP */
+
+const { normalizeArguments, kConfigurationError } = require('../utils');
+
+function HttpApi(transport, ConfigurationError) {
+  this.transport = transport;
+  this[kConfigurationError] = ConfigurationError;
+}
+
+HttpApi.prototype.request = function (method, params, options, callback) {
+  [params, options, callback] = normalizeArguments(params, options, callback);
+  if (Array.isArray(params.body)) {
+    params.bulkBody = params.body;
+    delete params.body;
```

Comment:
Is this modifying input? Meaning if I pass in `params` into the call twice, do I get a different `params` in the second call?